PxWeb/Code/Api2/DataSource/Cnmm/CnmmDataSource.cs

```diff
@@ -90,12 +96,83 @@ public CnmmDataSource(ICnmmConfigurationService cnmmConfigurationService, IItemS
             return null;
         }
 
+        public Codelist? GetCodelist(string id, string language)
+        {
+            Codelist? codelist = null;
+            var cnmmOptions = _cnmmConfigurationService.GetConfiguration();
+
+            if (string.IsNullOrEmpty(id))
+            {
+                return codelist;
+            }
+
+            if (id.StartsWith("agg_", System.StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Maybe splitt this into smaller private methods that you can call for readability.

Reply:
Made private methods to get CNMM Groupings and Valuesets